cmd/launcher/platform/platform.go

```diff
@@ -0,0 +1,22 @@
+package platform
+
+import (
+	"fmt"
+
+	"github.com/buildpacks/lifecycle/cmd/launcher/platform/common"
+	"github.com/buildpacks/lifecycle/cmd/launcher/platform/pre06"
+	v06 "github.com/buildpacks/lifecycle/cmd/launcher/platform/v06"
+	v07 "github.com/buildpacks/lifecycle/cmd/launcher/platform/v07"
+)
+
+func NewPlatform(apiStr string) (common.Platform, error) {
+	switch apiStr {
+	case "0.3", "0.4", "0.5":
+		return pre06.NewPlatform(apiStr), nil
```

Comment:
or maybe `vLegacy`. I agree the name `pre06` is kind of odd. Or maybe something silly like defining modules for each version that point to the v05.
```
v03 = v05.NewPlatform("0.3")
v04 = v05.NewPlatform("0.4")
```

I kind of wonder if instead of having `v06` have code like this:

```
func (p *v06Platform) AnalyzeOperations() []lifecycle.AnalyzeOperation {
	return []lifecycle.AnalyzeOperation{pre06.ReadPreviousImage, pre06.RestoreLayerMetadata}
}
```

We could always re-define the steps to make it obvious they are used as-is or modified
```
ReadPreviousImage = WithOptionalImage(pre06.ReadPreviousImage)
RestoreLayerMetadata = pre06.RestoreLayerMetadata

func (p *v06Platform) AnalyzeOperations() []lifecycle.AnalyzeOperation {
	return []lifecycle.AnalyzeOperation{ReadPreviousImage,RestoreLayerMetadata}
}
```

Reply:
It kind of bothers me having the "ops" function definitions in the `platform` package. I feel that the `lifecycle` package should sort of stand on its own, and someone should be able to import it in creating their own lifecycle binary (with the platform of their choosing).

Reply:
@yaelharel @jabrown85 please see 682f577

WRT the naming, I agree it's weird... maybe it's a bit better with 682f577 since we're not reaching back to `pre06` for any behavior. If we were to completely refactor all of our lifecycle code to use this pattern, we would end up creating `v03`, `v04`, and `v05` packages to get the differentiated behavior.

Reply:
I'm wondering if https://github.com/buildpacks/lifecycle/commit/682f5770e256fafdc67424f231046d6792e77151 is better than what you did before, because when, for example, platform API 15 will be released, some of the functions won't be relevant to most of the versions.
But even if we keep the last change as is, I think it's better to rename the package `pre06`.

Reply:
682f577 looks like a decent step forward to me. All the variations being next to each other is nice. I like that each version is standing on their own. It means deleting a version should not affect any other version.

Reply:
@yaelharel I agree with you, that was the reason I put it in `platform` to begin with... but after seeing how confusing it is to reference previous platforms, this seems like the lesser of two evils!

Reply:
Also hopefully one day we will deprecate and then remove support for some platforms!

Reply:
Please see 70d06dc - `v3through5` seems not bad...

Reply:
Per @aemengo - 

> I really dislike this package name v03through05. I think @jabrown85 suggested package legacy. What's wrong with that? Let's prioritize code that's easily understood, not code that tries too hard to teach you.

I'm also good with `legacy`.

Reply:
Does it make sense that we won't need to make a new package in one of the next versions? I guess not, but if there is a chance that how would we call the package that supports the two versions?

If there is no chance that such thing will happen, I'm totally fine with `legacy`, but if there is, maybe it's better to rename it to `v03Tov05` or `v03To05`.

Reply:
I think eventually every platform version will have its own package. The only reason we don't have `v03`, `v04`, and `v05` yet is because we didn't refactor the code that pertains to those packages to use the `Platform` interface. If we always use this interface going forward (when adding support for new platforms), we won't have another platform package that supports more than one platform.

Reply:
See 4ad354f - went with `legacy`